ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -135,28 +158,10 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	// include STOPPED which is also a terminal status because `suspend` requires to stop the Ray job gracefully before
 	// delete the RayCluster.
 	if isJobSucceedOrFail(rayJobInstance.Status.JobStatus) {
-		// If the function `updateState` updates the JobStatus to Complete successfully, we can skip the reconciliation.
```

Comment:
With this PR, we only delete RayCluster (TTL) if the `JobDeploymentStatus` is `JobDeploymentStatusComplete`. Hence, we don't need to check whether it is deleted or not if the status if not `JobDeploymentStatusComplete`.